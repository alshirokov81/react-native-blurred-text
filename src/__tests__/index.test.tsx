import React from 'react';
import renderer from 'react-test-renderer';
import { Platform } from 'react-native';
import RNBlurredText from '../index';

describe('<RNBlurredText />', () => {
  const testText = 'Blurred text';
  const textSize = 20;
  const blurrVal = 4;
  const textColor= 'rgb(20,200,220)';

  it('Should display text for android', () => {
    Platform.select=jest.fn().mockImplementation((obj) => obj.android);

    const testRenderer = renderer.create(
      <RNBlurredText
      textVal={testText}
      textSize={textSize}
      blurrVal={blurrVal}
      textColor={textColor}
      />
    );
    expect (testRenderer.toJSON()).toMatchSnapshot();
  });

  it('Should display text for ios', () => {
    Platform.select=jest.fn().mockImplementation((obj) => obj.ios);

    const testRenderer = renderer.create(
      <RNBlurredText
      textVal={testText}
      textSize={textSize}
      blurrVal={blurrVal}
      textColor={textColor}
      />
    );
    expect (testRenderer.toJSON()).toMatchSnapshot();
  });
});

