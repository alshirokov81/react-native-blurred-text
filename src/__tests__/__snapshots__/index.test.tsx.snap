// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RNBlurredText /> Should display text 1`] = `
<BlurredTextView
  blurrVal={4}
  textColor="rgb(20,200,220)"
  textSize={20}
  textVal="Blurred text"
/>
`;

exports[`<RNBlurredText /> Should display text for android 1`] = `
<BlurredTextView
  blurrVal={4}
  textColor="rgb(20,200,220)"
  textSize={20}
  textVal="Blurred text"
/>
`;

exports[`<RNBlurredText /> Should display text for ios 1`] = `
<BlurredTextView
  blurrVal={2.6}
  textColor="rgb(20,200,220)"
  textSize={20}
  textVal="Blurred text"
/>
`;
